src/utils/Assets.ts

```diff
@@ -0,0 +1,332 @@
+        const templateAddresses: any[] = Object.values(templatesAvailable)
+        index = templateAddresses.findIndex(function (item) {
+          return item === template
+        })
+      } else {
+        const templateIndexes = Object.keys(templatesAvailable)
+        index = templateIndexes.findIndex(function (item) {
+          return item.indexOf(template.toString()) !== -1
+        })
+      }
+      if (index !== -1) {
+        index += 1
+      }
+    }
+  }
+  // index or -1 if not found
+  return index
+}
+
+/**
+ * Use this function to accurately calculate the template index, and also checking if the template is active
+ * @param owner the signer account
+ * @param nftContractAddress the nft contract address, usually artifactsAddresses.ERC721Factory
+ * @param template the template ID or template address (from smart contract getId() function)
+ * @returns index of the template on the list
+ */
+export async function calculateActiveTemplateIndex(
+  owner: Signer,
+  nftContractAddress: string, // addresses.ERC721Factory,
+  template: string | number
+): Promise<number> {
+  // is an ID number?
+  const isTemplateID = typeof template === 'number'
+
+  const factoryERC721 = new NftFactory(nftContractAddress, owner)
+  const currentTokenCount = await factoryERC721.getCurrentTokenTemplateCount()
+  for (let i = 1; i <= currentTokenCount; i++) {
+    const tokenTemplate = await factoryERC721.getTokenTemplate(i)
+
+    const erc20Template = new ethers.Contract(
+      tokenTemplate.templateAddress,
+      ERC20Template.abi,
+      owner
+    )
+
+    // check for ID
+    if (isTemplateID) {
+      const id = await erc20Template.connect(owner).getId()
+      if (tokenTemplate.isActive && id.toString() === template.toString()) {
+        return i
+      }
+    } else if (
+      tokenTemplate.isActive &&
+      tokenTemplate.templateAddress === template.toString()
+    ) {
+      return i
+    }
+  }
+  // if nothing is found it returns -1
+  return -1
+}
+/**
+ *
+ * @param name asset name
+ * @param symbol asse symbol
+ * @param owner owner address
+ * @param assetUrl asset url
+ * @param templateIndex 1,2 or 4
+ * @param ddo ddo
+ * @param encryptDDO encrypt or not?
+ * @param providerUrl the provider URL
+ * @param providerFeeToken the provider fee token
+ * @param nftContractAddress the nft contract address
+ * @param aquariusInstance aquarius, could be node instance url
+ * @param dispenserAddress dispenser address
+ * @param fixedRateAddress fixed rate exchange address
+ * @param baseTokenAddress base token address (ocean)
+ * @returns ddo id as string
+ */
+export async function createAsset(
+  name: string,
+  symbol: string,
+  owner: Signer,
+  assetUrl: any,
+  template: string | number, // If string, it's template address , otherwise, it's templateId
+  ddo: any,
+  encryptDDO: boolean = true, // default is true
+  providerUrl: string,
+  providerFeeToken: string,
+  nftContractAddress: string, // addresses.ERC721Factory,
```

Comment:
change to nftFactory.
but why? We can get nftFactory from addresses(env or contracts npm package), so we don't need it as param